### ISSUE TITLE
End of path shows during animation

### ISSUE THREAD

**Issue by MichaelPriebe** [alexjlockwood/ShapeShifter#285]
[vector.txt](https://github.com/alexjlockwood/ShapeShifter/files/2107789/vector.txt) (change extension to .shapeshifter)

![](https://thumbs.gfycat.com/PeacefulAltruisticEeve-size_restricted.gif)

**Comment by alexjlockwood** [comment]
Hmm... very strange, because I don't see this behavior at all.

Could you give me some information about your environment? i.e. Browser version, OS version?

**Comment by MichaelPriebe** [comment]
I'm on Firefox, Windows 10. This extra part shows when I export it too. (Not sure if exporting relies on the browser rendering engine)

**Comment by MichaelPriebe** [comment]
Exporting on Chrome has the same effect (even though it doesn't show in the viewer)
![untitled1](https://user-images.githubusercontent.com/5238952/41501686-0f59e89a-7177-11e8-9d51-de59380f8499.gif)
You can see the dot appear, and disappear multiple times, and also as the check fades away it also splits up into multiple segments.

**Comment by alexjlockwood** [comment]
Could you copy/paste the contents of the `.shapeshifter` file to this bug report (for the second gif you posted... i.e. the one where it splits up into multiple segments as it fades away).

**Comment by alexjlockwood** [comment]
I'm able to reproduce in SVG spritesheet exports and on FireFox... seems like using `strokeLineCap="round"` maybe the culprit... hmm.

**Comment by tylerthurston11** [comment]
yeah of course! [Try this one](https://www.dropbox.com/s/5fv40vtirq703qg/v1E.shapeshifter?dl=0)

There's quite a few layers because we were trying out Shapeshifter for a logo animation idea we had. But even so, I tried starting a new project with only one layer added from importing an SVG and I seemed to be getting the same end appearing during the animation point issue.

Best of luck finding a solution Alex and thanks for making such a cool tool! Despite the hiccup we've still really enjoyed using it :)

**Comment by jamesraulw** [comment]
I opened the .svg file created by the Shape Shifter and I could see why this dot is shown. It's a result rounding mistake, probably. 
![Untitled-1](https://user-images.githubusercontent.com/57112022/67710681-29ec9e80-f99f-11e9-9acf-cc640a2fca56.jpg)

When I sum up the dash and gap value of the objects, I could see that the imperfect frame of the animation has +0.001 ou -0.001 than the perfect frame. Adjusting this values, I could fix the frame. 

This extra point is a restart of the stroke of the path that appear due to a math problem that doesn't match correctly the dash and gap value.

Sorry for my bad English. I hope I could help.

**Comment by bhaarata** [comment]
@MichaelPriebe , I think, that your path is wrong for some reason! Instead of `M 28 8.4 L 15.4 27.7 L 8.1 21` you should have `M8.1 21L15.4 27.7L28 8.4`. The start point in your file is on the end point and end point is on the start point.

You can close this issue, because it is your mistake.
